src/ens.ts

```diff
@@ -37,8 +37,8 @@ export default class Ens extends EthereumNamingService {
    * @param source - if specified as a string will be used as main url, if omited then defaults are used
    * @throws ConfigurationError - when either network or url is setup incorrectly
    */
-  constructor(source: NamingServiceSource = true, web3Provider?: Web3Provider) {
-    super(web3Provider);
+  constructor(source: NamingServiceSource = true, provider?: Provider) {
```

Comment:
The constructor misses documentation about `provider` argument.
Also we don't validate `provider` here, though we validate it in the `fromWeb3Provider` method. The behaviour should be consistent

Reply:
I guess this constructor assumes passing etherjs provider only, we should explicitly mention it in the documentation, cause it can be confusing for the users

Reply:
Ens.ts is for internal use only, and Resolution has all the constructors that deal with the conversion of providers to make sure it is consistent. I added the docs to it as well